src/main/scala/org/broadinstitute/dsde/firecloud/dataaccess/HttpGoogleServicesDAO.scala

```diff
@@ -272,4 +273,20 @@ object HttpGoogleServicesDAO extends GoogleServicesDAO with FireCloudRequestBuil
     val pipeline: HttpRequest => Future[GooglePriceList] = sendReceive ~> decode(Gzip) ~> unmarshal[GooglePriceList]
     pipeline(Get(FireCloudConfig.GoogleCloud.priceListUrl))
   }
+
+  def status: Future[SubsystemStatus] = {
+    try {
+      val storage = new Storage.Builder(httpTransport, jsonFactory, getBucketServiceAccountCredential).setApplicationName("firecloud").build()
+      val bucketResponse = storage.buckets().list(FireCloudConfig.FireCloud.serviceProject).executeUsingHead()
+      Future(
+        bucketResponse.getStatusCode match {
+          case x if x == 200 => SubsystemStatus(ok = true, messages = None)
+          case _ => SubsystemStatus(ok = false, messages = Some(List(bucketResponse.parseAsString())))
+        }
+      )
+    } catch {
+      case t: Throwable => Future(SubsystemStatus(ok = false, Some(List(t.getMessage))))
```

Comment:
I want to take a second look at this to let the exception bubble up and be handled by `StatusService`